app/javascript/articles/components/__tests__/SaveButton.test.jsx

```diff
@@ -0,0 +1,87 @@
+import { h } from 'preact';
+import { fireEvent, render } from '@testing-library/preact';
+import { axe } from 'jest-axe';
+import { SaveButton } from '../SaveButton';
+
+it('should have no a11y violations', async () => {
+  const article = { class_name: 'Article', id: 1 };
+  const { container } = render(<SaveButton article={article} />);
+  const results = await axe(container);
+
+  expect(results).toHaveNoViolations();
+});
+
+it('should render button as bookmarked', () => {
+  const article = { class_name: 'Article', id: 1 };
+  const { queryByText } = render(<SaveButton article={article} isBookmarked />);
+  const saveButton = queryByText('Saved');
+
+  expect(saveButton).not.toBeNull();
+});
+
+it('should button as not being bookmarked', () => {
+  const article = { class_name: 'Article', id: 1 };
+  const { queryByText } = render(
+    <SaveButton article={article} isBookmarked={false} />,
+  );
+  const saveButton = queryByText('Save');
+
+  expect(saveButton).not.toBeNull();
+});
+
+it('should bookmark when it previously was not', async () => {
```

Comment:
The only thing that needs a mock here in the next two tests @dengaku12 in regards to clicking the button is the `onClick` prop. Jest has a convenient `jest.fn()` which allows us to do this. For the actual click of the button we want that to be real, which we can do via `.click()`.